src/main/java/server/HTTProtocol.java

```diff
@@ -0,0 +1,39 @@
+package server;
+
+import java.io.BufferedReader;
+import java.io.PrintWriter;
+import java.util.Arrays;
+
+import static server.StatusCodes.OK;
+import static server.StatusCodes.NOT_FOUND;
+import static server.Routes.ROUTES;
+
+public class HTTProtocol implements Runnable {
+    private Client client;
+    private BufferedReader in;
+    private PrintWriter out;
+
+    public HTTProtocol(Client client) {
+        this.client = client;
+        this.in = client.getInputStreamReader();
+        this.out = client.getOutputStreamWriter();
+    }
+
+    public void run() {
+        RequestParser parser = new RequestParser(in);
+        Request request = new Request(parser.parse());
+        String responseStatus = getResponseStatus(request);
+
+        out.print(responseStatus);
+        out.flush();
+        client.closeSocket();
+    }
+
+    private String getResponseStatus(Request request) {
```

Comment:
This function probably shouldn't belong here. I'd suggest having a new class with this responsibility